lib/taste_tester/ssh_util.rb

```diff
@@ -15,27 +15,77 @@
 module TasteTester
   class SSH
     module Util
-      def ssh_base_cmd
-        jumps = TasteTester::Config.jumps ?
-          "-J #{TasteTester::Config.jumps}" : ''
+      def jumps
+        TasteTester::Config.jumps ? "-J #{TasteTester::Config.jumps}" : ''
+      end
+
+      def ssh_cmd_generator
+        if TasteTester::Config.ssh_cmd_template
+          base_gen_args = {
+            :user => TasteTester::Config.user,
+            :jumps => jumps,
+            :host => @host,
+          }
+          TasteTester::Config.ssh_cmd_template %
+            base_gen_args.update(TasteTester::Config.ssh_gen_args)
+        end
+      end
+
+      def ssh_gen_cmd
+        if TasteTester::Config.ssh_cmd_template
+          # we store this generated command inside a class variable
+          # so that we can directly refer to this while printing
+          # logs and error messages
+          @ssh_gen_cmd =
+            Mixlib::ShellOut.new(ssh_cmd_generator).run_command.stdout.chomp
+          @ssh_gen_cmd
+        end
+      end
+
+      def ssh_vanilla_cmd
         "#{TasteTester::Config.ssh_command} #{jumps} -T -o BatchMode=yes " +
           '-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no ' +
           "-o ConnectTimeout=#{TasteTester::Config.ssh_connect_timeout}"
       end
 
+      def ssh_base_cmd
```

Comment:
one of these includes the target, the other does not. So you have conditionals everywhere. Why not just have this include the target in both cases?

Reply:
yeah - it was bit annoying since I was trying to minimize the churn. The thing will be much smoother if the base command just takes care of targets as well. I think tunnel.rb might need to change a bit for that...